### DISCUSSION TITLE
Personal access token not working.

### DISCUSSION THREAD

**Discussion by danish-khan-I** [community/community#58573] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Personal access token classic and fine-grained both stopped working for me. I tried re-generating but that didn't work. thought routine fix is happening then tried after 4-5hrs but still not working.

```git fetch
Username for 'https://github.com': username
Password for 'https://username@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/repo/repo/'
```

what could be the reason for that. I have been doing this over a year now.

**Comment by danish-khan-I** [comment]
I tried `curl -v -H "Authorization: token <your token>" https://api.github.com/user/issues` and its working. I am doing git pull/fetch on server directly. didn't want to use my ssh key there.

**Comment by DSDmark** [comment]
Hi @danish-khan-I,

Confirm first your token is not expired: https://github.com/settings/tokens

**Reply by danish-khan-I** [reply]
I already have mentioned I re-regenerated one and tried curl and it’s working.

**Comment by airtower-luna** [comment]
In my experience the "Support for password authentication was removed" message appears when whatever was entered isn't a valid token. If the same token works with `curl`, my best guess is that *something* about how you enter the token changed. How do you enter it? :thinking:

**Reply by danish-khan-I** [reply]
I just copy paste it. I also verified it by pasting it onto console to check if it has white-spaces but its intact. can you try it out and see if its working for you?

**Reply by danish-khan-I** [reply]
> I just copy paste it. I also verified it by pasting it onto console to check if it has white-spaces but its intact. can you try it out and see if its working for you?

I am using classic token so there’s no way I have permission issues but I tried fine grained too even different repos but the result was same.

**Reply by airtower-luna** [reply]
I just tried with a token of my own, works flawlessly. This may be an absurd question, but… In the log you posted, the username is `username`. I assume that's redacted and you use the actual username matching the token there? :sweat_smile:

**Reply by danish-khan-I** [reply]
I replaced it with my username ofc. 😅

**Comment by danish-khan-I** [comment]
Strange thing happened, I tried in different server and git clone/fetch worked there with same token. what could be the reason for it not working on the initial machine.

**Reply by danish-khan-I** [reply]
It worked on the initial machine as well. I am loosing my mind now. what's going unless someone from Github team confirm it was not working from their end.😢

**Reply by Frank-efg** [reply]
Did you find a fix for this? Currently stuck as well.

**Reply by baybay-creator** [reply]
Stuck as well

**Comment by MrLucifer-92** [comment]
Check the network settings. The expiration date and access levels also need to be checked.

**Comment by MrLucifer-92** [comment]


**Reply by MrLucifer-92** [reply]
Check the network settings. The expiration date and access levels also need to be checked.

**Comment by NH89** [comment]
**Sometimes you just have to wait and try again another day.**
I had the same problem. I was trying to push a commit my own github remote repository.
It kept refusing the token, even though I had a current classic token and was using the same procedure that I have for the last two years.
I generated three new tokens, none of them worked.
24 hrs later I tried again, and it just worked as if there had been no problem.

**Reply by MrLucifer-92** [reply]
Great

**Comment by GunmanCode** [comment]
Same problem here. Added `--force` flag and it worked.
